### PR TITLE
refactor useAsset, fix form problems

### PR DESCRIPTION
**Pull request by kdembler**
Turns out the fix for the problem with thumbnail updates in video sheet form we encountered in #848 was rather trivial after all. However I thought there was a problem with implementation of `useAsset` and did a refactor only to discover the problem is somewhere else. However, I think the refactored version is quite more readable so decided to open a PR for it anyway. This should also fix all the issues with the form.

### CODE REVIEW

**Comment by drillprop** [comment]
Working much better now. Unfortunately, there is still one problem left, which wasn't there before:

If you change the thumbnail of the video and publish changes you'll not see the updated image in the `MyVideosView`. When you click again on the edited video, you'll also see the old thumbnail
asset bug

**Comment by drillprop** [comment]
@kdembler I opened PR to your branch. It should resolve the issue, not sure if you like how I've done that, let me know what you think.

File: src/hooks/useAsset/useAsset.tsx [hunk 1]
```diff
@@ -1,120 +1,60 @@
...
+export const useAsset: UseAsset = ({ entity, assetType }, opts = {}) => {
   const { getStorageProvider } = useStorageProviders()
   const [error, setError] = useState<ErrorEvent | null>(null)
-  const [isLoading, setIsLoading] = useState<boolean | undefined>(undefined)
+  const [isLoading, setIsLoading] = useState<boolean>(true)
   const [url, setUrl] = useState<string | undefined>(undefined)
-  const cachedAssetData = useRef({})
+  const [cachedAssetData, setCachedAssetData] = useState<AssetData | null | undefined>(undefined)
+
+  const assetData = entity ? readAssetData(entity, assetType) : null
```
Comment (drillprop) [inline src/hooks/useAsset/useAsset.tsx]:
```suggestion
  const assetData = readAssetData(entity, assetType) 
```
I don't think it's needed - if an entity is null `readAssetData` returns null anyway

File: src/hooks/useAsset/useAsset.tsx [hunk 2]
```diff
@@ -1,120 +1,60 @@
...
+  const assetDataNotChanged = entity ? isEqual(assetData, cachedAssetData) : false
 
   useEffect(() => {
-    if (error) {
-      Logger.error(`Failed to load ${assetType}`, error)
+    if (!entity || assetDataNotChanged || opts.skip) {
```
Comment (drillprop) [inline src/hooks/useAsset/useAsset.tsx]: I could be wrong, but do we actually need `opts.skip`? I removed that from the code and didn't notice any difference, everything should work just fine. Tested updating thumbnails and switching tabs, with no problems

**Review by drillprop — COMMENTED** [review]
Found two small things in the code. Besides that everything works really well.

File: src/hooks/useAsset/useAsset.tsx (hunk 1 above)
Comment (drillprop) [reply src/hooks/useAsset/useAsset.tsx]: pretty much the same applies to the line underneath

File: src/hooks/useAsset/useAsset.tsx (hunk 2 above)
Comment (kdembler) [reply src/hooks/useAsset/useAsset.tsx]: Yeah seems it wasn't doing us any good, removed

File: src/hooks/useAsset/useAsset.tsx (hunk 1 above)
Comment (kdembler) [reply src/hooks/useAsset/useAsset.tsx]: 👍 removed

**Pull request merged**